src/walk.rs

```diff
@@ -48,13 +49,26 @@ pub fn scan(root: &Path, pattern: Arc<Regex>, config: Arc<FdOptions>) {
     let (tx, rx) = channel();
     let threads = config.threads;
 
+    let mut override_builder = OverrideBuilder::new(root);
+
+    for pattern in config.exclude_patterns.iter() {
+        let res = override_builder.add(pattern);
+        if res.is_err() {
+            error(&format!("Error: malformed exclude pattern '{}'", pattern));
+        }
+    }
+    let overrides = override_builder.build().expect(
```

Comment:
Any reason we should not use:
```rust
let overrides = override_builder.build().unwrap_or_else(|_| {
    error("Mismatch in exclude patterns")
});
```
instead?

Reply:
Thank you for the suggestion! That's better.